### PR TITLE
updated `for` loop to `for..of` in `AudioBuffer`

### PR DESCRIPTION
**Pull request by PrajwalBorkar**
#### Related issues

Part of #18103 
#### Metadata

- [ ] Adds a new document
- [ ] Rewrites (or significantly expands) a document
- [ ] Fixes a typo, bug, or other error

### CODE REVIEW

File: files/en-us/web/api/audiobuffer/index.md [hunk 1]
```diff
@@ -56,10 +56,10 @@ const myArrayBuffer = audioCtx.createBuffer(2, audioCtx.sampleRate * 3, audioCtx
 for (let channel = 0; channel < myArrayBuffer.numberOfChannels; channel++) {
```
Comment (PrajwalBorkar) [inline files/en-us/web/api/audiobuffer/index.md:56]: In this case how to identify if it is an array or not because it is using length as numberOfChannels?

**Comment by PrajwalBorkar** [comment]
Tried one case to check for errors. Please give suggestions about different cases for future reference. Thank you

File: files/en-us/web/api/audiobuffer/index.md (hunk 1 above)
Comment (teoli2003) [reply files/en-us/web/api/audiobuffer/index.md:56]: In this case, you can't use `for...of`. You need a `length` property and an accessor (the `[]` operator). AudioBuffer doesn't have them, so a traditional `for (;;)` is needed.

File: files/en-us/web/api/audiobuffer/index.md [hunk 2]
```diff
@@ -56,10 +56,10 @@ const myArrayBuffer = audioCtx.createBuffer(2, audioCtx.sampleRate * 3, audioCtx
...
   // This gives us the actual array that contains the data
   const nowBuffering = myArrayBuffer.getChannelData(channel);
-  for (let i = 0; i < myArrayBuffer.length; i++) {
+  for (const arrayVal of myArrayBuffer) {
     // Math.random() is in [0; 1.0]
     // audio needs to be in [-1.0; 1.0]
-    nowBuffering[i] = Math.random() * 2 - 1;
+    nowBuffering[arrayVal] = Math.random() * 2 - 1;
```
Comment (teoli2003) [inline files/en-us/web/api/audiobuffer/index.md:62]: This is incorrect, here you need the index, but `for...of` gives you a value. So, here again, you should keep the traditional `for(;;)` loop.

**Comment by teoli2003** [comment]
You got unlucky here, both cases couldn't be changed into a `for...of`, even if one used `length` (and triggered my search!).

`for...of` simplifies quite a few for loops, but not all of them. There will be some false positive.

Let's try another one. (Closing this PR)

**Pull request closed without merging**